exif/exif.go

```diff
@@ -27,6 +27,24 @@ const (
 	interopPointer = 0xA005
 )
 
+// A decodeError is returned when the image cannot be decoded as a tiff image.
+type decodeError struct {
```

Comment:
exactly. we don't want to break other goexif users who have made the unfortunate decision to rely on error message strings.